commands/create_cluster_test.go

```diff
@@ -128,106 +179,114 @@ func Test_CreateClusterSuccessfully(t *testing.T) {
 			    ]
 			  }
 			]`))
-		}
-	}))
-	defer mockServer.Close()
-
-	var testCases = []addClusterArguments{
-		// minimal arguments
-		{
-			apiEndpoint: mockServer.URL,
-			owner:       "acme",
-			token:       "fake token",
-		},
-		// extensive arguments
-		{
-			apiEndpoint:         mockServer.URL,
-			clusterName:         "UnitTestCluster",
-			numWorkers:          4,
-			releaseVersion:      "0.3.0",
-			owner:               "acme",
-			token:               "fake token",
-			workerNumCPUs:       3,
-			workerMemorySizeGB:  4,
-			workerStorageSizeGB: 10,
-			verbose:             true,
-		},
-		{
-			apiEndpoint:   mockServer.URL,
-			clusterName:   "Cluster Name from Args",
-			owner:         "acme",
-			token:         "fake token",
-			inputYAMLFile: "testdata/minimal.yaml",
-			verbose:       true,
-		},
-	}
-
-	validateErr := errors.New("")
-	executeErr := errors.New("")
+			}
+		}))
 
-	cmdAPIEndpoint = mockServer.URL
-	initClient()
+		cmdAPIEndpoint = mockServer.URL
+		cmdToken = testCase.inputArgs.token
+		initClient()
 
-	for i, testCase := range testCases {
-		validateErr = validateCreateClusterPreConditions(testCase)
+		validateErr = validateCreateClusterPreConditions(*testCase.inputArgs)
 		if validateErr != nil {
-			t.Errorf("Validation error in testCase %v: %s", i, validateErr.Error())
+			t.Errorf("Validation error in testCase %d: %s", i, validateErr.Error())
 		}
-		_, executeErr = addCluster(testCase)
+		_, executeErr = addCluster(*testCase.inputArgs)
 		if executeErr != nil {
-			t.Errorf("Execution error in testCase %v: %s", i, executeErr.Error())
+			t.Errorf("Execution error in testCase %d: %s", i, executeErr.Error())
 		}
+
+		mockServer.Close()
```

Comment:
moving this to line 184 right after creating `mockServer` as `defer mockServer.Close()` would be safer

Reply:
See https://github.com/giantswarm/gsctl/pull/262